### ISSUE TITLE
bug: telemetry deployment fails

### ISSUE THREAD

**Issue by kaytharius** [Azure/terraform-azurerm-lz-vending#283]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Versions

Please paste the output of `terraform version` command from within the initialized directory:
Terraform v1.6.4
on windows_amd64
+ provider registry.terraform.io/azure/azapi v1.10.0
+ provider registry.terraform.io/hashicorp/azurerm v3.80.0

Please enter the module version that you are using:

```text
module.lz_vending.azapi_resource.telemetry
```

### Description
I am attempting to authenticate with a service principal and a certificate but am getting an error when the telemetry module is reached and the azapi provider is used. It doesn't seem to use the service principal configuration in the azapi provider block and throws an error asking me to run "az login". I have run "az account show" and "Get-AZContext". Both show that I am not logged in. The plan makes it through the subscription creation and registering of providers prior to the error. 

│ Error: reading "Resource: (ResourceId \"/subscriptions/<my-subscription>/providers/Microsoft.Resources/deployments/pid-50a8a460-d517-4b11-b86c-6de447806b67_3.4.0_00000003\" / Api Version \"2021-04-01\")": ChainedTokenCredential: failed to acquire a token.
│ Attempted credentials:
│       managed identity timed out
│       AzureCLICredential: ERROR: Please run 'az login' to setup account.
│
│
│   with module.lz_vending.azapi_resource.telemetry_root[0],
│   on .terraform\modules\lz_vending\main.telemetry.tf line 3, in resource "azapi_resource" "telemetry_root":
│    3: resource "azapi_resource" "telemetry_root" {

#### Steps to Reproduce

1. Using the following provider blocks:
provider "azurerm" {
  features {}

  client_id                   = var.client_id
  client_certificate_path     = var.client_certificate_path
  client_certificate_password = var.client_certificate_password
  tenant_id                   = var.tenant_id
  subscription_id             = var.subscription_id
}

provider "azapi" {
  client_id                   = var.client_id
  client_certificate_path     = var.client_certificate_path
  client_certificate_password = var.client_certificate_password
  tenant_id                   = var.tenant_id
 #subscription_id             = var.subscription_id
}

2. run "terraform plan"

**Comment by kaytharius** [comment]
After further testing, I must have been logged in during my original deployment of the subscription and the registering of the providers. I am now seeing this when registering the providers. I have the same credentials defined in the azapi provider block as I do in the azurerm block.

**Comment by matt-FFFFFF** [comment]
This sounds like an authentication issue with AzAPI.

Can you try a test without using this module? Create a simple resource like a resource group using AzAPI.

**Comment by kaytharius** [comment]
The issue ended up being with the cert. I used the documentation to create the cert on Windows with OpenSSL but apparently azapi was unable to read it.

I found the following in the debug log.

2023-12-06T10:13:04.518-0600 [INFO]  provider.terraform-provider-azapi_v1.10.0.exe: newDefaultAzureCredential failed to initialize environment credential:
	failed to load certificate from "C:\service-principal.pfx": pkcs12: unknown digest algorithm: 2.16.840.1.101.3.4.2.1: timestamp=2023-12-06T10:13:04.518-0600
	
After recreating the cert with the following options, I was able to authenticate. 

openssl pkcs12 -certpbe PBE-SHA1-3DES -keypbe PBE-SHA1-3DES -export -macalg sha1  .......

This is where I found the solution.
https://discuss.hashicorp.com/t/azure-service-principal-client-certificate-error/32037?msclkid=9351d13bd12411ec96deb763dd60b7af
